src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs

```diff
@@ -147,15 +157,27 @@ protected override void ApplyResponseChallenge()
                 properties = new AuthenticationProperties(ChallengeContext.Properties);
             }
 
-            if (string.IsNullOrEmpty(properties.RedirectUri))
+            string redirectToUse = null;
```

Comment:
@PinpointTownes in review, I think it is confusing. I scaled back to just the 'local' redirect. I think part of the confusion is that Option.RedirectUri has the same name as AuthenticationProperties.RedirectUri but they are used for different purposes. Let's deal with dynamic 'redirect_uri' separately.